posts/colliding-with-quarkus.md

```diff
@@ -0,0 +1,91 @@
+---
+authors:
+- Gerard Gigliotti
+tags:
+date: 2019-09-10T17:11:00.000Z
+title: "Colliding with Quarkus"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2019/09/quarkus.png
+---
+
+Recently at [Voxxed Days Melbourne](https://australia.voxxeddays.com/) there was a Red Hat booth, and they were handing out [Quarkus](https://quarkus.io/) stickers. Obviously if it's good enough to put on a sticker, it must be important. So, what is Quarkus? According to their website, Quarkus is a Cloud Native, (Linux) Container First framework for writing Java applications. It's really a mix of two tightly integrated components: a Java Framework and GraalVM.
+
+![Laptop with Stickers](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2019/09/quarkusLaptop.png)
+
+## Java Framework
+On first impressions, the framework seems very similar to SpringBoot, which isn't a bad thing at all. At its core, it supports all the dependency injection features you could want, based on [this](http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html). From there, Quarkus combines a list of curated extensions to allow you to easily bolt on new functionality as required. If you need to include a REST server, you can include the `quarkus-resteasy` dependency. If you need access to AWS' DynamoDB, there is ` quarkus-amazon-dynamodb`. You don't need to go hunting for dependencies or versions to use; all the important stuff is available for you to pull in via the `listExtensions` command.
+
+## GraalVM
+[GraalVM](https://www.graalvm.org/) is a relatively new universal virtual machine for running JVM-based languages, as well as JavaScript, Python, Ruby and R. The feature that Quarkus really embraces is the [SubstrateVM](https://github.com/oracle/graal/tree/master/substratevm), which allows Java applications to be compiled into native executable images. Instead of running your Java code against a JVM, you can compile it down to a portable, native executable. Quarkus abstracts all this complexity from you, and allows you to just run `nativeImage` to build a executable binary.
+
+# Taking it for a spin
+So I have put together a [small HTTP server example](https://github.com/ggotti/kanye-quarkus) which basically just proxies requests to the `https://api.kanye.rest` service, which returns a random Kanye West quotes. The application has three small classes:
```

Comment:
either remove "a" or change "quotes" to "quote"

Reply:
👍